test/src/error-snippets/utils.ts

```diff
@@ -0,0 +1,203 @@
+import path from "path";
+import sinon from "sinon";
+import url from "url";
+import fs from "fs-extra";
+import {
+    softFileURLToPath,
+    formatFileNameHeader,
+    getSourceCodeFile,
+    formatErrorSnippet,
+} from "../../../src/error-snippets/utils";
+import type { codeFrameColumns } from "@babel/code-frame";
+
+const codeFrame = require("@babel/code-frame"); // eslint-disable-line @typescript-eslint/no-var-requires
+
+const withGray = (line: string): string => "\x1B[90m" + line + "\x1B[39m";
+
+describe("error-snippets/utils", () => {
+    const sandbox = sinon.createSandbox();
+
+    afterEach(() => sandbox.restore());
+
+    describe("softFileURLToPath", () => {
+        it("should return same fileName if it does not start with 'file://'", () => {
+            const file = "data/myFile.txt";
+
+            const result = softFileURLToPath(file);
+
+            assert.equal(result, file);
+        });
+
+        it("should convert file URL to path if fileName starts with 'file://'", () => {
+            const file = "file:///data/myFile.txt";
+
+            const result = softFileURLToPath(file);
+
+            assert.equal(result, "/data/myFile.txt");
+        });
+
+        it("should return same fileName if it starts with 'file://' but conversion fails", () => {
+            sandbox.stub(url, "fileURLToPath").throws(new Error("foo"));
+            const file = "file://invalid/path.txt";
+
+            const result = softFileURLToPath(file);
+
+            assert.equal(result, file);
+        });
+    });
+
+    describe("formatFileNameHeader", () => {
+        it("should handle line Offset correctly", () => {
+            const fileName = "myFile.txt";
+
+            const opts = {
+                line: 10,
+                linesAbove: 5,
+                linesBelow: 5,
+            };
+
+            const result = formatFileNameHeader(fileName, opts);
+            const expected = ["   . | // myFile.txt", "   . |"]
```

Comment:
for `expected` value it would be better to use plain string instead of generation it